File: 3731-FindMissingElements/javascript/solution.js
// Last updated: 3/6/2026, 12:38:51 AM
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findMissingElements = function(nums) {
    var max = nums[0];
    var min = nums[0];
    let result = [];
    for(let i of nums){
        if(i>max){
            max = i;
        }
        if(i<min){
            min = i;
        }
    }
    for(let i = min; i<=max; i++){
        if(nums.includes(i)==false){
            result.push(i);
        }
    }
    return result;
};

File: 3731-FindMissingElements/cpp/solution.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<int> findMissingElements(vector<int>& nums) {
        if (nums.empty()) return {};
        int max = nums[0];
        int min = nums[0];
        vector<int> result;
        for (int i : nums) {
            if (i > max) {
                max = i;
            }
            if (i < min) {
                min = i;
            }
        }
        unordered_set<int> numSet(nums.begin(), nums.end());
        for (int i = min; i <= max; i++) {
            if (numSet.find(i) == numSet.end()) {
                result.push_back(i);
            }
        }
        return result;
    }
};


int main(){
    
    return 0;
}